### ISSUE TITLE
[Accordion] Disabled accordion items are not announced as 'disabled'/'dimmed'

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3153]
### Provide a general summary of the issue here

##Current behavior 

When user place their focus on `sl-accordion-item` with `disabled` attribute screen readers announce that element the same as if it was active.
User doesn't get information about item being disabled or dimmed.
This happens with VoiceOver on Firefox Safari. With NVDA this happens only on Firefox.

<img width="708" height="239" alt="Image" src="https://github.com/user-attachments/assets/34670c8b-da2f-424c-877a-819121d9d449" />

<img width="690" height="309" alt="Image" src="https://github.com/user-attachments/assets/50f4bc33-4f9b-4d59-81cc-cc20cec3fb3a" />

## Suggested solution 

 For better a11y support on Firefox and Safari it's recommended to add `aria-disabled` to disabled `sl-accordion-item` so screen readers would announce that the element is 'unavailabled'/'dimmed'.
It's then recommended to remove `disabled` attribute and replace its function with JS, CSS and "tabindex='-1'" 

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
